go/ioutil2/ioutil.go

```diff
@@ -20,11 +20,16 @@ func WriteFileAtomic(filename string, data []byte, perm os.FileMode) error {
 		return err
 	}
 	n, err := f.Write(data)
+	if err == nil {
```

Comment:
I went back and read the doc for syncfs & close. It says that close does not have to sync. But I still think it's safe. I think the importance of syncfs is only relevant for data loss due to power failure scenarios.
However, they say you have to check the return value of f.Close because an error there means that a previous write could have failed.